### PR TITLE
add gofmt step to ci

### PR DESCRIPTION
**Pull request by wataru-nakanishi**
PR for https://github.com/roc-streaming/roc-go/issues/86

Implemented
- [x]  Add gofmt CI step

This is my first contributing!
If the contributing process was wrong, please tell me.

### CODE REVIEW

**Comment by wataru-nakanishi** [comment]
@roc-streaming/go ready for review

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -71,6 +71,12 @@ jobs:
         with:
           go-version: ${{ matrix.go }}
 
+      - name: gofmt
```
Comment (wataru-nakanishi) [inline .github/workflows/build.yaml:74]: I added fmt step only in linux.
Because Jerome1337/gofmt-action@v1.0.5 is available in linux only.

**Comment by gavv** [comment]
Thanks for PR! Using linux is perfectly ok, we don't need to check formatting on each OS because results would be the same. 

I suggest extracting formatting check into a separate job "Formatting". It's not really bound to any steps in "Linux" job and can be done separately.

**Comment by gavv** [comment]
Follow-up issue: https://github.com/roc-streaming/roc-go/issues/92

**Comment by gavv** [comment]
@wataru-nakanishi Could you please check if this CI step will fail if a file has no newline at the end? If it won't, I think I'll create a separate issue for adding such step too.

**Comment by wataru-nakanishi** [comment]
@gavv 
Thank you!
Fixed.

> Could you please check if this CI step will fail if a file has no newline at the end? If it won't, I think I'll create a separate issue for adding such step too.

I added a new line to config.go and CI failed.
Originally, there are no new line
https://github.com/roc-streaming/roc-go/actions/runs/5035614704/jobs/9031245933?pr=91

**Comment by gavv** [comment]
> I added a new line to config.go and CI failed.
Originally, there is no new line


I actually mean the opposite case: not when extra newline is added, but when the last one is missing.

**Comment by wataru-nakanishi** [comment]
@gavv 
CI check didn't fail if a file has no newline at the end.
For example, config.go has no new line.
https://github.com/roc-streaming/roc-go/blob/main/roc/config.go#L494

Please let me know if I'm misunderstanding.

**Comment by gavv** [comment]
config.go _does_ have newline at the end. It's last symbol is LF, not "}". If there were no terimnating LF, github would display that red icon that I showed above.

**Comment by wataru-nakanishi** [comment]
@gavv 
> config.go does have newline at the end. It's last symbol is LF, not "}". If there were no terimnating LF, github would display that red icon that I showed above.

Thanks for letting me know.
CI failed if a file has no new line.
https://github.com/roc-streaming/roc-go/actions/runs/5035923474/jobs/9031753001?pr=91

**Comment by gavv** [comment]
Thanks!! So we don't need one more check, gofmt already checks everything we need.

One more comment: could you please place formatting job before release job, and add formatting job to the list of dependencies of release job?

**Comment by wataru-nakanishi** [comment]
@gavv 
> could you please place formatting job before release job, and add formatting job to the list of dependencies of release job?

fixed! [38053ff](https://github.com/roc-streaming/roc-go/pull/91/commits/38053ff2950aab249decad1ebe2f72e74efd2ac5)

**Comment by gavv** [comment]
LGTM

**Pull request merged**